lib/Basics/ReadWriteLock.cpp

```diff
@@ -130,6 +130,20 @@ void ReadWriteLock::lockRead() {
   }
 }
 
+bool ReadWriteLock::tryLockReadFor(std::chrono::microseconds timeout) {
+  auto end = std::chrono::high_resolution_clock::now() + timeout;
+  if (tryLockRead()) {
+    return true;
+  }
+  std::unique_lock<std::mutex> guard(_reader_mutex);
+  while (true) {
+    if (tryLockRead()) {
+      return true;
+    }
+    _readers_bell.wait_until(guard, end);
+  }
+  return false;
+}
```

Comment:
Is it expected that return false is impossible?

Reply:
Yes

Reply:
It's a bug.

Reply:
It will be busy loop if timeout will be reached

Reply:
Valery is right. The `while (true)` does not honor the timeout. You have to use a similar approach as in `tryLockWriteFor` with the `cv_status`